programs/restaking/src/modules/fund/commands/cmd11_harvest_restaking_yield.rs

```diff
@@ -1048,9 +1054,17 @@ impl HarvestRestakingYieldCommand {
 
                 drop(fund_account);
 
-                // Update pricing
+                // Update pricing & high water mark
                 FundService::new(ctx.receipt_token_mint, ctx.fund_account)?
-                    .new_pricing_service(accounts.iter().copied(), true)?;
+                    .update_asset_values(&mut pricing_service, true)?;
+
+                let mut fund_account = ctx.fund_account.load_mut()?;
+                let one_receipt_token_as_sol_after_token_compounded =
+                    fund_account.one_receipt_token_as_sol;
+
+                fund_account.fee_harvested_one_receipt_token_as_sol +=
+                    one_receipt_token_as_sol_after_token_compounded
+                        - one_receipt_token_as_sol_before_token_compounded;
```

Comment:
can we add a simple assertion here?: `one_receipt_token_as_sol_after_token_compounded-one_receipt_token_as_sol_before_token_compounded > 0`.

I see the entire logic is sound and complete. It will be just a guard for unexpected behavior if by any chance.